shivyc/parser/declaration.py

```diff
@@ -252,10 +252,21 @@ def parse_parameter_list(index):
     if token_is(index, token_kinds.close_paren):
         return params, index
 
+    # argument is void and is not function declaration
+    if token_is(index, token_kinds.void_kw) \
+            and token_is(index + 1, token_kinds.close_paren):
+        if token_is(index + 2, token_kinds.open_brack):
+            return params, index + 1
+        elif token_is(index + 1, token_kinds.identifier):
```

Comment:
I'm a little confused by this code. Isn't the token at `index + 1` always going to be a close parenthesis here? 

I think it would be easier to implement this in the semantic stage of the compiler (i.e. in the `tree/ `directory) rather than in the parser, since this isn't really a parsing issue as much as it's a semantic one.

Reply:
Fix this on new pull request #102